Sources/Datadog/Core/Upload/DataUploadConditions.swift

```diff
@@ -30,6 +30,13 @@ internal struct DataUploadConditions {
     }
 
     private func shouldUploadFor(batteryStatus: BatteryStatus) -> Bool {
+        if batteryStatus.state == .unknown {
+            // Note: in RUMS-132 we got the report on `.unknown` battery state reporing `-1` battery level on iPad device
+            // plugged to Mac through lightning cable. As `.unkown` may lead to other unreliable values,
+            // it seems safer to arbitrary allow uploads in such case.
+            return true
```

Comment:
it makes more sense to return false only in certain cases and return true for all the rest. that way, we will be uploading by default

```swift
let runningOnBattery = (state == .unplugged)
let lowBattery = runningOnBattery && (level <= .minBattery)
let lowPowerEnabled = .isLowPowerEnabled
return !(lowBattery || lowPowerEnabled)
```

Reply:
What's the difference 💭? It seems similar, with the conditions being just re-written, but I don't find it easier for read.

And it doesn't have the same behaviour, for `state = .unkown` + `.isLowPowerModeEnabled = true` it gives `false`. As I suggested in the PR description, IMHO we shouldn't rely on any further info if the state is `.unknown`.

Reply:
🎗️ for the records: `UIDevice.BatteryState.{all unknown cases}` are mapped to `Datadog.BatteryState.unknown` and `.unknown` here is `Datadog.BatteryState.unknown`. basically, `Datadog.BatteryState` is a frozen enum in that case.